FrontEnd/src/BookingPage/BookingPage.jsx

```diff
@@ -31,46 +33,62 @@ function BookingForm({ userData, socket }) {
     vendorUID,
     vendorUsername,
     appointmentData,
-    userData
+    userData,
+    vendorEmail,
+    vendorName,
+    postID
   ) => {
     await requestAppointment(
       userUID,
       vendorUID,
       vendorUsername,
       appointmentData,
-      userData
+      userData,
+      vendorEmail,
+      vendorName,
+      postID
     );
   };
 
   useEffect(() => {
-    const isFormValid = Object.values(appointmentData).every(
-      (value) => value !== ""
-    );
+    const isFormValid = Object.keys(appointmentData).every(key => {
+      if (key === 'appointmentAdditionalNote') {
+        return true; // Skip validation for optional input
+      }
+      return appointmentData[key] !== '';
+    });
+
     setIsFormValid(isFormValid);
-    if (userUID === post.userId) {
+
+    if (userUID === post.vendorUID) {
       setIsFormValid(false);
     }
-  }, [appointmentData]);
+  }, [appointmentData, userUID, post.vendorUID]);
+
 
   const handleSubmit = async (event, type) => {
     try {
       event.preventDefault();
-      if (userUID === post.userId) {
+      if (userUID === post.vendorUID) {
```

Comment:
UI-side feedback: instead of let user book there own appointment and reject their request, it may be better to directly disable the book button when user is booking their own appointments.